packages/gateway/src/gateway-fetch-metadata.ts

```diff
@@ -47,10 +49,9 @@ export class GatewayFetchMetadata {
   async getCredits(): Promise<GatewayCreditsResponse> {
     try {
       const baseUrl = new URL(this.config.baseURL);
-      const creditsUrl = `${baseUrl.origin}/v1/credits`;
 
       const { value } = await getFromApi({
-        url: creditsUrl,
+        url: `${baseUrl.origin}/v1/credits`,
```

Comment:
No need for `baseUrl`, see https://github.com/vercel/ai/pull/8750#discussion_r2361335889

```suggestion
        url: `${this.config.baseURL}/v1/credtis`
```